index.js

```diff
@@ -28,3 +29,20 @@ async function fetchCharacters() {
 }
 
 fetchCharacters();
+
+nextButton.addEventListener("click", () => {
+  if (page <= 42) {
+    page += 1;
+    fetchCharacters();
+    pagination.textContent = `${page} / 42`;
+    //console.log("page: ", page);
+  }
+});
```

Comment:
The logic itself is correct. Now you just need to use the variable `maxPage` here instead of a hardcoded value.

Please remove the `console.log()`.
And as you update the text content of your pagination in the event listener of both your buttons, you could instead move this line to the fetch function - cause then you need it just once. After you have fetched the data, you can update the text content of the pagination.